src/server/journal/types.h

```diff
@@ -3,45 +3,52 @@
 //
 #pragma once
 
+#include <string>
+#include <variant>
+
 #include "server/common.h"
 #include "server/table.h"
 
 namespace dfly {
 namespace journal {
 
-enum class Op : uint8_t {
+enum class OpCode : uint8_t {
+  // 0-20 control opcodes
   NOOP = 0,
-  LOCK = 1,
-  UNLOCK = 2,
-  LOCK_SHARD = 3,
-  UNLOCK_SHARD = 4,
-  SCHED = 5,
-  VAL = 10,
-  DEL,
-  MSET,
+
+  // 21-50 commands
+  SET = 21,
+  DEL = 22,
+  LPUSH = 23,
 };
 
-// TODO: to pass all the attributes like ttl, stickiness etc.
-struct Entry {
-  Entry(Op op, DbIndex did, TxId tid, std::string_view skey)
-      : opcode(op), db_ind(did), txid(tid), key(skey) {
-  }
+using KeyType = std::string_view;
+using ValueType = std::variant<KeyType, const PrimeValue*>;
+using ListType = ArgSlice;
 
-  Entry(DbIndex did, TxId tid, std::string_view skey, const PrimeValue& pval)
-      : Entry(Op::VAL, did, tid, skey) {
-    pval_ptr = &pval;
-  }
+using PldEmpty = std::tuple<>;
```

Comment:
it requires an explanation and some comments. I assume that each type fits a group of commands but I can not make the connection on how  you model each command arguments.